src/complete.cpp

```diff
@@ -211,9 +211,9 @@ completion_t::completion_t(wcstring comp, wcstring desc, string_fuzzy_match_t ma
       flags(resolve_auto_space(completion, flags_val)) {}
 
 completion_t::completion_t(const completion_t &) = default;
-completion_t::completion_t(completion_t &&) = default;
+completion_t::completion_t(completion_t &&) = default;  // NOLINT
```

Comment:
I assume this lint is about adding `noexcept`, which I believe is pointless since we disable exceptions. I get something similar for `lru_node_t` from `clangd`:

`src/lru.h:36:9: warning:Move constructors should be marked noexcept (fix available)`

Can we disable this lint somewhere in the project configuration? I guess clang-tidy/clangd could detect `-fno-exceptions`.
If not, we could perhaps add an explicit `NOLINT(performance-noexcept-move-constructor)`, but that's quite verbose, I'm not sure. `// NOLINT: noexcept` would work too I guess.

Reply:
Currently, all performance checks are enabled: https://github.com/fish-shell/fish-shell/blob/8d2e4bbef58645e51271ae36a4f99ea4d7c5b709/.clang-tidy#L2

Reply:
Right, I disabled it, so the `// NOLINT` should not be necessary.

Reply:
Removed.

Reply:
Oh sorry I have been to hasty. 
Turns out we still need to mark them `noexcept` for the STL to use them, as stated in http://clang.llvm.org/extra/clang-tidy/checks/performance-noexcept-move-constructor.html, example at https://godbolt.org/z/vzEWcc.

4c08cbd050de7488adfcad000ef2b1b61f26be7c actually does this. Now the question is, why not just add `noexcept` to this line as well? I remember there was some problem, maybe a build failure on an old system?

Reply:
I think so. I remember the buildbots failing on it.

Reply:
Added noexcept. Let's see if anything fails.